### PR TITLE
CU-472 field improvements

### PR DESCRIPTION
**Pull request by ralfzen**
- Every model now has as a PK UUID
- Categories can be filtered now by `is_global` for having Categories throughout all organizations and also new organizations available
- The ProductType is now a model for adding more fields to the type later
- These ProductTypes must be related to Category
- Added a filter for the Product.part_number

*this PR is related to the previous PR because of the migrations*

### CODE REVIEW

File: products/models.py [hunk 1]
```diff
@@ -21,12 +23,22 @@ def __str__(self):
         return self.name
 
 
+class ProductType(models.Model):
+    uuid = models.UUIDField(primary_key=True, default=uuid.uuid4)
+    name = models.CharField(max_length=255, help_text="Type name")
+    category = models.ForeignKey(ProductCategory, related_name='types', on_delete=models.CASCADE)
```
Comment (docktorrr) [inline products/models.py]: We decided to have tree-like Category model for this, it's more universal and fits well the current needs. So type will be level 1 category and category will be level 0 category.

File: products/permissions.py [hunk 2]
```diff
@@ -23,8 +23,26 @@ def has_object_permission(self, request, view, obj):
...
             else:
                 raise PermissionDenied('User is not in the same organization as the object.')
+
+
+class ProductTypeOrganizationPermission(OrganizationPermission):
+    def has_object_permission(self, request, view, obj):
+        if request.user.is_superuser:
+            return True
+
+        if obj.category.is_global:
```
Comment (docktorrr) [inline products/permissions.py]: We don't need actually this permission class, we can use OrganizationPermission for this, just to add new condition there.

File: products/permissions.py [hunk 3]
```diff
@@ -23,8 +23,26 @@ def has_object_permission(self, request, view, obj):
...
             if request.session.get('jwt_organization_uuid') == str(obj.organization_uuid):
                 return True
             else:
                 raise PermissionDenied('User is not in the same organization as the object.')
+
+
+class ProductTypeOrganizationPermission(OrganizationPermission):
+    def has_object_permission(self, request, view, obj):
+        if request.user.is_superuser:
+            return True
```
Comment (docktorrr) [inline products/permissions.py]: We don't need actually this permission class, we can use OrganizationPermission for this, just to add new condition there.

File: products/models.py (hunk 1 above)
Comment (ralfzen) [reply products/models.py]: we agreed on adding the `level` - field later because we don't need it for the beginning

**Pull request merged**